pkgs/os-specific/cygwin/newlib-cygwin/default.nix

```diff
@@ -0,0 +1,127 @@
+{
+  lib,
+  stdenvNoLibc,
+  newlib-cygwin-headers,
+  buildPackages,
+  automake,
+  autoconf,
+  autoreconfHook,
+  bison,
+  cocom-tool-set,
+  flex,
+  perl,
+  w32api,
+}:
+
+stdenvNoLibc.mkDerivation {
+  pname = "newlib-cygwin";
+
+  inherit (newlib-cygwin-headers)
+    version
+    src
+    meta
+    ;
+
+  outputs = [
+    "out"
+    "bin"
+    "dev"
+    "man"
+  ];
+
+  patches =
+    newlib-cygwin-headers.patches
+    ++ [
+      # https://cygwin.com/pipermail/cygwin-developers/2020-September/011970.html
+      # This is required for boost coroutines to work. After we get to the point
+      # where nix runs on cygwin, we can attempt to upstream this again.
+      ./store-tls-pointer-in-win32-tls.patch
+    ]
+    ++ lib.optional (stdenvNoLibc.hostPlatform != stdenvNoLibc.buildPlatform) ./fix-cross.patch;
```

Comment:
Does this break the build if it's not cross? It's generally a good idea to unconditionally apply patches.

Reply:
I can't be 100% sure it breaks it until we get cygwin hosted builds, but IMO it's too hacky to use unconditionally.  I'd rather assume it's not needed for now, and see where we end up.  We can clean up and upstream after that.

Reply:
Sounds good.